docs/design.md

```diff
@@ -0,0 +1,29 @@
+# Overarching thoughts (North Star)
+* The primary purpose for Blocks is to be an easier environment for students to program robots.
+    * Students should be able to program in Blocks without any experienced software mentor help.
+    * This means that reducing confusion in blocks is more important than being 1:1 mapping to Python
+    * For ease of use, we are ok with not being able to do everything in Blocks you can do in Python.
+* You should be able to make a competitive robot using Blocks
+    * Our goal is to rasie the floor, not lower the ceiling
+* Programming in Blocks is still programming.   You can make logic errors and programs that don't work and have errors
+* Some students will move on from Blocks to text based languages
+    * Students that want to do more than they can in Blocks can export from Blocks and use that as a starting point for their own Python coding.
+
+# Intentional Limitations (for simplicity)
+* Each blockly workspace goes to one Python file (module) and can only contain one class
+* The only type of methods that can be created are class methods
+* All variables created in blockly will be class instance variables
+
+# Design thoughts
+## OpModes
+* An OpMode can be either Autonomous or Teleop
+* An OpMode can be marked as Disabled or Enabled
+* An OpMode has access to the Robot class
+## Robot
+* There should be one robot class that is shared between all OpModes
+* A robot has zero or more mechanisms 
```

Comment:
What does "has" mean here? Does it mean that the robot class has an instance variable for each of its mechanisms?

Reply:
I was thinking it would be something like:
self.mechanisms as a dict with the name of the mechanism as the key and the instance of the class as the value.

Reply:
That sounds good.